zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -152,6 +152,45 @@ sealed trait Schema[A] {
 }
 
 object Schema extends TupleSchemas with RecordSchemas with EnumSchemas {
+
+  private def fuzzyReformat(str: String): String = str.replaceAll("[-_]", "").toLowerCase
```

Comment:
nit: `String.replaceAll` will compile the regex on every invocation. Probably want to define a static regex and reuse it. So something like:

```scala
val reformatPattern = Pattern.compile("[-_]")

def fuzzyReformat(str: String): String = reformatPattern.matcher(str).replaceAll("")
```